client/src/components/SharedComponents.jsx

```diff
@@ -1,7 +1,8 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
-const StarRating = ({ rating }) => {
+const StarRating = ({ rating = 0, clickable = false, setRating = null }) => {
```

Comment:
The level of logic and modification required justify an independent component, given the time and inclination.